pkgs/applications/misc/joshuto/default.nix

```diff
@@ -0,0 +1,31 @@
+{ fetchFromGitHub, lib, rustPlatform, stdenv, SystemConfiguration }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "joshuto";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "kamiyaa";
+    repo = pname;
+    rev = version;
+    sha256 = "08d6h7xwcgycw5bdzwwc6aaikcrw3yc7inkiydgml9q261kql7zl";
+    # upstream includes an outdated Cargo.lock that stops cargo from compiling
+    postFetch = ''
+      mkdir -p $out
+      tar xf $downloadedFile --strip=1 -C $out
+      substituteInPlace $out/Cargo.lock \
+        --replace 0.8.6 0.9.0
```

Comment:
```suggestion
        --replace 0.8.6 ${version}
```

Reply:
doesn't really matter since it is a one time change but ok